rest_framework-stubs/serializers.pyi

```diff
@@ -81,13 +81,57 @@ class BaseSerializer(Generic[_IN], Field[Any, Any, Any, _IN]):
     instance: _IN | None
     initial_data: Any
     _context: dict[str, Any]
-    def __new__(cls: type[Self], *args: Any, **kwargs: Any) -> Self: ...
     def __class_getitem__(cls, *args, **kwargs): ...
-    def __init__(
-        self,
+    # When both __init__ and __new__ are present, mypy will prefer __init__
+    @overload
+    def __new__(
+        cls: type[Self],
+        instance: Iterable[_IN] | None = ...,
+        data: Any = ...,
+        partial: bool = ...,
+        many: Literal[True] = ...,
+        allow_empty: bool = ...,
+        context: dict[str, Any] = ...,
+        read_only: bool = ...,
+        write_only: bool = ...,
+        required: bool = ...,
+        default: Any = ...,
+        initial: Any = ...,
+        source: str = ...,
+        label: str = ...,
+        help_text: str = ...,
+        style: dict[str, Any] = ...,
+        error_messages: dict[str, str] = ...,
+        validators: Sequence[Validator[Any]] | None = ...,
+        allow_null: bool = ...,
+    ) -> ListSerializer[_IN]: ...
```

Comment:
I don't think there's a way of inferring the correct subclass of `ListSerializer`, as it is [defined](https://www.django-rest-framework.org/api-guide/serializers/#customizing-listserializer-behavior) in the `Meta` class